ui/src/AzureOutputChannel.ts

```diff
@@ -0,0 +1,50 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { OutputChannel, ViewColumn } from "vscode";
+
+export class AzureOutputChannel implements OutputChannel {
+    public readonly name: string;
+    private _outputChannel: OutputChannel;
+
+    constructor(name: string, outputChannel: OutputChannel) {
+        this.name = name;
+        this._outputChannel = outputChannel;
+    }
+
+    public append(value: string): void {
+        this._outputChannel.append(value);
+    }
+
+    public appendLine(value: string): void {
+        this._outputChannel.appendLine(value);
+    }
+
+    public appendLog(value: string, resourceName?: string, date?: Date): void {
```

Comment:
Can you refactor `resourceName` and `date` into an `options` object? There's several examples in `index.d.ts`.

I want to avoid things like this:
```
appendLog(value, undefined, 'today')
```

And instead do something like this:
```
appendLog(value, { date: 'today' })
```

It's not too bad with only two parameters, but you can imagine something ugly like this if you ever add support for more options:
```
appendLog(value, undefined, undefined, undefined, 'otherProp')
```